### PR TITLE
add burn test #197

### PR DESCRIPTION
**Pull request by bizzyvinci**
## Related Issue
Which issue does this pull request resolve?
#197 

## Description of changes
Add test to confirm that burn outside of lock would revert. This is because lockedBy.length is 0 and therefore an underflow error in [_accountDelta](https://github.com/Uniswap/v4-core/blob/d8c1ceed2a029cef8a9249f799c2a996bfa27107/contracts/PoolManager.sol#L183)

> Secondly what if an unsafe transfer is done

unsafe transfer is [impossible](https://forum.openzeppelin.com/t/unsafe-transfer-in-erc1155/18758) cause OZ implementation strictly complies with the [ERC1155 spec](https://eips.ethereum.org/EIPS/eip-1155).

### CODE REVIEW

File: contracts/test/PoolBurnTest.sol [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+    function lockAcquired(uint256, bytes calldata rawData) external returns (bytes memory) {
+        require(msg.sender == address(manager));
+
+        CallbackData memory data = abi.decode(rawData, (CallbackData));
+
+        if (data.isMint) {
+            if (data.amount0 > 0) {
+                uint256 currencyBalBefore = IERC20Minimal(Currency.unwrap(data.key.currency0)).balanceOf(data.sender);
+                uint256 managerBalBefore = manager.balanceOf(data.sender, data.key.currency0.toId());
```
Comment (emmaguo13) [inline contracts/test/PoolBurnTest.sol:44]: Might be more clear + consistent (with our other tests) to check the balance of the 1155 in the forge test file like this [line](https://github.com/Uniswap/v4-core/blob/571c7941914b1fb05521e31300a0068a4612358b/test/foundry-tests/PoolManager.t.sol#L895)

**Review by emmaguo13 — COMMENTED** [review]
Would be a good idea to look into native eth support as well

**Comment by bizzyvinci** [comment]
Eth support has been added and ERC1155 balance check has been moved to PoolManager.t.sol
[Balance check](https://github.com/bizzyvinci/uniswap-v4-core/blob/67d29d40c66cd272ec85805350b219e949702e59/contracts/test/PoolBurnTest.sol#L64) for ETH is different because before and after balance are the same due to `msg.value`

**Comment by snreynolds** [comment]
Closing as burning outside of a lock is no longer an issue/not allowed.

**Pull request closed without merging**